cryptography/src/lib.rs

```diff
@@ -131,12 +131,29 @@ pub trait BatchScheme: Specification {
 /// (which should be cheap to clone).
 pub trait Digest: Array + Copy {}
 
-/// An object that can generate a digest.
-pub trait Digestible<D: Digest>: Clone + Send + Sync + 'static {
-    /// Generate a digest from the object.
+/// An object that can be uniquely represented as a [Digest].
+pub trait Digestible<D: Digest>: Clone + Sized + Send + Sync + 'static {
+    /// Returns the unique representation of the object as a [Digest].
+    ///
+    /// If many objects can be represented by the same [Digest], you
+    /// should implement [Identifiable] instead.
     fn digest(&self) -> D;
 }
 
+/// An object that can be identified by a [Digest].
+///
+/// When possible, crates should prefer supporting [Identifiable] over [Digestible] (
+/// returning all [Digestible] objects for a given `identity()`).
+pub trait Identifiable<D: Digest>: Clone + Sized + Send + Sync + 'static {
```

Comment:
We should make Identifiable Extend Digest.

Reply:
extend `Digestible` you mean?

Reply:
If we did extend, it would I think imply "Digests" of the same format. That might be a bit strong of an assumption.